### PR TITLE
[mle] delay to send announce for FTD

### PR DESCRIPTION
**Pull request by librasungirl**
This commit delays sending Announce for FTD after it has become Router and decided to stay in REED on the safe side:
OT device would try to attach on the announced channel only once, and if it fails, the device would switch back to old channel. However the FTD may get a different router id, or it will take longer for REED to become router especially in linear topology, causing the attaching device fail to attach and may no chance to attach again on the announced channel.

### CODE REVIEW

File: src/core/thread/mle_router.cpp [hunk 1]
```diff
@@ -4311,6 +4316,9 @@ void MleRouter::HandleAddressSolicitResponse(Coap::Header *aHeader, Message *aMe
         mRouters[GetLeaderId()].SetNextHop(GetRouterId(mParent.GetRloc16()));
     }
 
+    // send announce after become Router if needed
+    InformPreviousChannel();
```
Comment (jwhui) [inline src/core/thread/mle_router.cpp]: Move this to below `exit:` in case anything in handling the address response fails?

Comment (librasungirl) [reply src/core/thread/mle_router.cpp]: Updated, Thanks.

File: src/core/thread/mle.cpp [hunk 2]
```diff
@@ -670,6 +665,22 @@ otError Mle::SetStateChild(uint16_t aRloc16)
     return OT_ERROR_NONE;
 }
 
+void Mle::InformPreviousChannel(void)
+{
+    VerifyOrExit(mPreviousPanId != Mac::kPanIdBroadcast);
```
Comment (jwhui) [inline src/core/thread/mle.cpp]: Should we validate that the device is attached?

Comment (librasungirl) [reply src/core/thread/mle.cpp]: Updated. Thanks.

**Comment by jwhui** [comment]
Is there a case where we need to send MLE Announce messages on the previous channel after becoming a Leader?

**Comment by librasungirl** [comment]
I could not think of such scenarios. For my understanding, the device switches to the new channel to attach when hearing MLE announce on its original channel, that is, there have already been one partition on the new channel. Additionally, if the device fails to attach on the new channel,  it won't try to become leader, but switch back to its original channel.
Thoughts?

**Pull request merged**